### PR TITLE
JAMES-1950 Track execution time in the mailets

### PR DESCRIPTION
**Pull request by chibenwa**
Because we are blind with the mailets, and I believe we would gain much knowledge on our application from this... I added time tracking on the mailet pipeline, and on what I believe are the more time consuming mailets. 

I added the following metrics : 
 - "spoolProcessing" tracking the execution times for the mailet pipeline
 - "localDeliveryTimeMetric" for LocalDelivery
 - "rrtExecution" for recipient rewriting
 - "sieveExecution" for tracking Sieve execution time
 - "remoteDeliveryAttempt" for tracking remoteDelivery time
 - "vacationMailetExecution" for tracking vacation mailet execution time

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1950-3-a71d6147-5ff4-4b3c-930c-7521aecefa54] Results :
[Building Java 8 james-1950-3-a71d6147-5ff4-4b3c-930c-7521aecefa54] 
[Building Java 8 james-1950-3-a71d6147-5ff4-4b3c-930c-7521aecefa54] Tests in error: 
[Building Java 8 james-1950-3-a71d6147-5ff4-4b3c-930c-7521aecefa54]   CassandraModSeqProviderTest.nextModSeqShouldGenerateUniqueValuesWhenParallelCalls:101 » Completion
```

**Comment by chibenwa** [comment]
test this please

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/Sieve.java [hunk 1]
```diff
@@ -84,8 +88,13 @@ public void init() throws MessagingException {
 
     @Override
     public void service(Mail mail) throws MessagingException {
-        List<MailAddress> recipientsWithSuccessfulSieveExecution = executeRetrieveSuccess(mail);
-        mail.setRecipients(keepNonDiscardedRecipients(mail, recipientsWithSuccessfulSieveExecution));
+        TimeMetric timeMetric = metricFactory.timer("sieveExecution");
```
Comment (quynhn) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/Sieve.java]: To make it's consistent, extract "sieveExecution" to the constant

**Comment by mbaechler** [comment]
why can't we have metrics on any pipeline stages instead ?

**Comment by chibenwa** [comment]
Agree with you @mbaechler , thanks for remark.

I will update CamelProcessor for doing this.

**Pull request closed without merging**